ZukoAzula2016/src/org/team1540/zukoazula/Shooter.java

```diff
@@ -25,20 +26,38 @@
     private static final BehaviorArbitrator rollerArb = new BehaviorArbitrator("Rollers");
     private static final ArbitratedFloat rollerSpeed = rollerArb.addFloat();
 
+    private static final TalonExtendedMotor intakeArmRollerCAN = FRC.talonCAN(7);
+
     public static void setup() throws ExtendedMotorFailureException {
         rollerSpeed.send(FRC.talonSimpleCAN(8, FRC.MOTOR_FORWARD));
 
         BangBangTalon shooter = new BangBangTalon(makeLinkedTalons(), "Shooter");
 
-        // passively roll to the wheel in teleop, while the shooter is stopped.
-        rollerSpeed.attach(rollerArb.addBehavior("Passive Roll", FRC.inTeleopMode().and(shooter.isStopped)), ZukoAzula.mainTuning.getFloat("Roller Passive Speed", 0.1f));
-
         PauseTimer actuallyFiring = new PauseTimer(ZukoAzula.mainTuning.getFloat("Shooter Fire Delay", 0.5f));
         // TODO: take robot modes into account
         BooleanInput warmup = ZukoAzula.controlBinding.addBoolean("Shooter Warm-Up");
         BooleanInput fire = ZukoAzula.controlBinding.addBoolean("Shooter Fire");
         BooleanInput spinup = warmup.or(fire).or(actuallyFiring);
 
+        EventInput intakeArmRollerForward = ZukoAzula.controlBinding.addEvent("Intake Arm Rollers Forward");
+        EventInput intakeArmRollerBackward = ZukoAzula.controlBinding.addEvent("Intake Arm Rollers Backward");
+        EventInput intakeArmRollerStop = ZukoAzula.controlBinding.addEvent("Intake Arm Rollers Stop");
+
+        StateMachine states = new StateMachine(0, "passive", "forward", "backward");
+        states.setStateWhen("passive", intakeArmRollerStop.or(spinup.onPress()).or(FRC.startTele));
+        states.setStateWhen("forward", intakeArmRollerForward.andNot(spinup));
+        states.setStateWhen("backward", intakeArmRollerBackward.andNot(spinup));
```

Comment:
You might be able to get away with just two BooleanCells, but I'm not certain whether it would be better or not long-term. Specifically, a BooleanCell for running and one for direction. (This wouldn't have the desynch issue or the imperative issue.)